js/lib/ajax.js

```diff
@@ -102,6 +102,11 @@ elgg.ajax.handleOptions = function(url, options) {
  * @private
  */
 elgg.ajax.handleAjaxError = function(xhr, status, error) {
+	if (!xhr.getAllResponseHeaders()) {
```

Comment:
After this gets in, can you make a PR merging 1.12 into 2.0? Best to handle conflicts early.